src/utils/web.py

```diff
@@ -143,7 +166,19 @@ def getUrlFd(url, headers=None, data=None, timeout=None):
         else:
             request = url
             request.add_data(data)
-        fd = urlopen(request, timeout=timeout)
+        if context is not None and ca_file is False and nocert is True:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is False and nocert is False:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is True and nocert is True:
+            fd = urlopen(request, timeout=timeout, context=context)
+        elif context is not None and ca_file is True and nocert is False:
+            try:
+                fd = urlopen(request, timeout=timeout)
+            except:
```

Comment:
Usually it's better to explicitly specify exceptions to catch.

Reply:
It's mandatory here, because urlopen takes time, so this `except` block is likely to catch `KeyboardInterrupt`.

Reply:
Actually, that was changed in Python 2.5: https://docs.python.org/2/library/exceptions.html#exception-hierarchy

KeyboardInterrupt now inherits from BaseException instead of Exception